src/Shared/VolumeNameGenerator.cs

```diff
@@ -20,8 +20,10 @@ public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suf
             throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. Only [a-zA-Z0-9_.-] are allowed.", nameof(suffix));
         }
 
-        // Create volume name like "myapplication-postgres-data"
-        var applicationName = builder.ApplicationBuilder.Environment.ApplicationName;
+        // Create volume name like "{sha256}-postgres-data"
+
+        // Compute a short hash of the content root path to differentiate between multiple AppHost projects with similar volume names
+        var applicationName = builder.ApplicationBuilder.Configuration["AppHost:Sha256"]![..10].ToLowerInvariant();
```

Comment:
As a user, how am I supposed to know which volume maps to which app in Docker Desktop?

Reply:
You don't know the SHA256 off the top of your head?!? I was waiting for feedback to say we should keep the assembly name as part of the prefix. I think instead of what it does today around dropping the name if the characters are invalid, we should instead remove those characters.

Reply:
Is there a length limit? Maybe we put the 5-10 character SHA at the end of the volume name?

Reply:
I think the limit is OS specific (these end up being paths). I also think we should lowercase the name.

Reply:
> I also think we should lowercase the name.

I don't necessarily object to that. I just want to be able to understand what each volume is used for.

Reply:
Yeah I think we should keep the application name in here and append the SHA to disambiguate.

Reply:
Also you'll need to update the doc-comment to reflect whatever you end up making the format.

Reply:
Doc comments on internals APIs 👀

Reply:
Then delete it 😄

Reply:
Done